aten/src/ATen/native/mps/operations/LinearAlgebra.mm

```diff
@@ -624,14 +615,23 @@ Tensor addbmm_mps(const Tensor& self, const Tensor& batch1, const Tensor& batch2
   }
 
   checkInputsSolver(A, B, left, "linalg.solve_triangular");
-  Tensor A_, B_;
-  std::tie(B_, A_) = _linalg_broadcast_batch_dims(B, A, /*don't check errors*/nullptr);
-  at::native::resize_output(out, B_.sizes());
+  Tensor A_t, B_t;
+  std::tie(B_t, A_t) = _linalg_broadcast_batch_dims(B, A, /*don't check errors*/nullptr);
+  at::native::resize_output(out, B_t.sizes());
 
   if (A.numel() == 0 || B.numel() == 0 || out.numel() == 0) {
+    out.zero_();
     return out;
   }
 
+  Tensor A_ = A_t;
+  Tensor B_ = B_t;
+  if (A_t.is_view() || !A_t.is_contiguous()) {
```

Comment:
I don't think the is_view is needed here, we can solve directly views  (storage offset) directly
```suggestion
  if (!A_t.is_contiguous()) {
```